bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/values/PercentageValue.java

```diff
@@ -154,7 +154,7 @@ public String getMQTTpublishValue(@Nullable String pattern) {
 
     @Override
     public StateDescriptionFragmentBuilder createStateDescription(boolean readOnly) {
-        return super.createStateDescription(readOnly).withMaximum(max).withMinimum(min).withStep(step)
+        return super.createStateDescription(readOnly).withMaximum(HUNDRED).withMinimum(BigDecimal.ZERO).withStep(step)
```

Comment:
The `step` now seems wrong to me.
When the we have `new PercentageValue (0,1000, 100, null, null)`, then we do not want a step of `100` here, right?
I think, the step here should be `stepPercent `

Reply:
I think you're right, but it didn't work for me at first try. 

I think we'll need to define / document how the percentage type is expected to behave and what the fields are defining.And then review / adapt all the code to get this behaviour.



On a side note: when doing a custom transformation with JavaScript, I first needed to have 0..1 as range for the input (to get 0..254 as output range), which I could not explain, but then after a restart it was 0..100 like expected, so for me there are still some strange things / things I don't understand going on with the percentage type...

Reply:
There are also a lot of things unclear to me going from the `Value` towards the UI.
I would want to cleanup the values in the sense, that we separate the changes from the UI from those comming from the device.
Currently a change from the UI is going through `ChannelState.publishValue`, which then calls `update` on the value.
A change from the device is going through `ChannelState.processMessage', which then also calls `update`on the value.
Therefor in `update` there is no way to know, from where the change originated. So setting an absolute value can cause a problem, because the base [min...max] or [0...100] is not clear.